doc/man1/openssl-pkeyutl.pod.in

```diff
@@ -210,8 +242,9 @@ hex dump the output data.
 
 =item B<-asn1parse>
 
-Parse the ASN.1 output data, this is useful when combined with the
-B<-verifyrecover> option when an ASN1 structure is signed.
+Parse the ASN.1 output data to check its DER encoding and print any errors.
+When combined with the B<-verifyrecover> option, this may be useful only in case
+an ASN.1 DER-encoded structure had been signed directly (without hashing it).
```

Comment:
But that's isn't true for PKCS#1 v1.5 where the signed hash is DER encoded along with its OID (and NULL parameter set):

```
$ openssl pkeyutl -verifyrecover -in rsa-sig.dat -inkey rsa.pem -asn1parse
    0:d=0  hl=2 l=  49 cons: SEQUENCE          
    2:d=1  hl=2 l=  13 cons:  SEQUENCE          
    4:d=2  hl=2 l=   9 prim:   OBJECT            :sha256
   15:d=2  hl=2 l=   0 prim:   NULL              
   17:d=1  hl=2 l=  32 prim:  OCTET STRING      
      0000 - c4 b9 47 51 ff c8 69 5a-06 a1 0f c8 e3 2b e3 4c   ..GQ..iZ.....+.L
      0010 - 83 75 f0 5e ce 0e b2 94-26 d6 01 f5 cd ee a1 1f   .u.^....&.......
```
The signature was a standard RSA + sha256 signature.

Reply:
And the new "help" comment for `-asn1parse` is I think too narrow in scope, it isn't just for syntax checks, it just decodes the DER output.  Could be done instead as a `| openssl asn1parse -inform DER`, but it does save a couple of keystrokes.

Reply:
Good point that in this case the signature has a DER encoding.

Yet as this PR here just backports another one that is already merged where this issue was not spotted,
we should not fix this here but generally do this for all versions.
I've  meanwhile prepared a further PR  #25958 
fixing this and several other remaining `pkeyutl` doc issues, part of which you had noted earlier, e.g., here: https://github.com/openssl/openssl/pull/25831#discussion_r1834084150